Common/DCAFitter/include/DCAFitter/FwdDCAFitterN.h

```diff
@@ -1256,6 +1269,26 @@ void FwdDCAFitterN<N, Args...>::print() const
   LOG(info) << "Stopping condition: Max.param change < " << mMinParamChange << " Rel.Chi2 change > " << mMinRelChi2Change;
   LOG(info) << "Discard candidates for : Rvtx > " << getMaxR() << " DZ between tracks > " << mMaxDXIni;
 }
+//___________________________________________________________________
+template <int N, typename... Args>
+inline bool FwdDCAFitterN<N, Args...>::propagateToVtx(o2::track::TrackParCovFwd& t, const std::array<float, 3>& p, const std::array<float, 2>& cov) const
+{
+  // propagate track to vertex including MCS effects if material budget included, simple propagation to Z otherwise
+  if (mUseMatBudget) {
+    float x2x0 = 0;
+    if (mTGeoFallBackAllowed) {
+      auto geoMan = o2::base::GeometryManager::meanMaterialBudget(t.getX(), t.getY(), t.getZ(), p[0], p[1], p[2]);
+      x2x0 = (float)geoMan.meanX2X0;
+    } else {
+      auto mb = mMatLUT->getMatBudget(t.getX(), t.getY(), t.getZ(), p[0], p[1], p[2]);
+      x2x0 = (float)mb.meanX2X0;
+    }
```

Comment:
I think the logic for the fall-back should be the opposite: TGeo must be used if MatLUT is not available:
```
if (mMatLUT) {
    auto mb = mMatLUT->getMatBudget(t.getX(), t.getY(), t.getZ(), p[0], p[1], p[2]);
    x2x0 = (float)mb.meanX2X0;
} else if (TGeoFallBackAllowed) {
    auto geoMan = o2::base::GeometryManager::meanMaterialBudget(t.getX(), t.getY(), t.getZ(), p[0], p[1], p[2]);
    x2x0 = (float)geoMan.meanX2X0;
} 
```

Reply:
Fixed, thanks